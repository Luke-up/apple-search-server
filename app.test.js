//Supertest is used to create CRUD operations
const request = require("supertest");
const app = require("./app");

//test checks if response is recieved from server
describe("Unit test", () => {
  test("server should return response", (done) => {
    request(app)
      .get("/okay")
      .expect(200)
      .expect((res) => {
        res.body = "okay";
      })
      .end((err, res) => {
        if (err) return document(err);
        return done();
      });
  });
});

//test checks if the itunes API is functioning
describe("Snapshot test", () => {
  test("should carry out get request", async () => {
    const response = await request(app).get(
      "/media?term=Lucy&&media=movie&&attribute=movieTerm&&limit=1"
    );
    expect(response.body).toMatchSnapshot(this);
  });
});
