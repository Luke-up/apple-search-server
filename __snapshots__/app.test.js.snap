// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot test should carry out get request 1`] = `
Object {
  "resultCount": 1,
  "results": Array [
    Object {
      "artistName": "Luc Besson",
      "artworkUrl100": "https://is3-ssl.mzstatic.com/image/thumb/Video2/v4/e2/28/99/e228995f-dea7-8846-8a44-db891ed29209/pr_source.lsr/100x100bb.jpg",
      "artworkUrl30": "https://is3-ssl.mzstatic.com/image/thumb/Video2/v4/e2/28/99/e228995f-dea7-8846-8a44-db891ed29209/pr_source.lsr/30x30bb.jpg",
      "artworkUrl60": "https://is3-ssl.mzstatic.com/image/thumb/Video2/v4/e2/28/99/e228995f-dea7-8846-8a44-db891ed29209/pr_source.lsr/60x60bb.jpg",
      "collectionArtistId": 345353262,
      "collectionArtistViewUrl": "https://itunes.apple.com/us/artist/universal-studios-home-entertainment/345353262?uo=4",
      "collectionCensoredName": "Atomic Blonde / Lucy",
      "collectionExplicitness": "notExplicit",
      "collectionHdPrice": 24.99,
      "collectionId": 1430227835,
      "collectionName": "Atomic Blonde / Lucy",
      "collectionPrice": 19.99,
      "collectionViewUrl": "https://itunes.apple.com/us/movie/lucy/id898085303?uo=4",
      "contentAdvisoryRating": "R",
      "country": "USA",
      "currency": "USD",
      "discCount": 1,
      "discNumber": 1,
      "hasITunesExtras": true,
      "kind": "feature-movie",
      "longDescription": "From the visionary director of La Femme Nikita and The Professional and starring Scarlett Johansson and Academy AwardÂ® winner Morgan Freeman comes an action-thriller about a woman accidentally caught in a dark deal who turns the tables on her captors. Altered by a dangerous new drug allowing her to use 100% of her brain capacity, Lucy transforms into a merciless warrior evolved beyond human logic.",
      "previewUrl": "https://video-ssl.itunes.apple.com/itunes-assets/Video122/v4/15/12/d5/1512d5d3-4851-00f9-0298-bfb7423bf5d3/mzvf_3166177837682568005.640x354.h264lc.U.p.m4v",
      "primaryGenreName": "Action & Adventure",
      "releaseDate": "2014-07-25T07:00:00Z",
      "shortDescription": "From La Femme Nikita and The Professional to The Fifth Element, writer/director Luc Besson has",
      "trackCensoredName": "Lucy",
      "trackCount": 2,
      "trackExplicitness": "notExplicit",
      "trackHdPrice": 14.99,
      "trackHdRentalPrice": 3.99,
      "trackId": 898085303,
      "trackName": "Lucy",
      "trackNumber": 2,
      "trackPrice": 14.99,
      "trackRentalPrice": 3.99,
      "trackTimeMillis": 5361152,
      "trackViewUrl": "https://itunes.apple.com/us/movie/lucy/id898085303?uo=4",
      "wrapperType": "track",
    },
  ],
}
`;
